pkg/controllers/deprovisioning/validation.go

```diff
@@ -98,6 +111,19 @@ func (v *Validation) IsValid(ctx context.Context, cmd Command) (bool, error) {
 	return isValid, nil
 }
 
+func (v *Validation) getProhibitedCandidates(ctx context.Context) (sets.String, error) {
+	// Re-compute Maintenance Windows for candidate validation
+	mws, err := buildMaintenanceMap(ctx, v.kubeClient, v.clock)
```

Comment:
I feel like we are repeating this logic _a lot_. Is there a way to cut down on the number of times that we call basically the same sequence of functions?

Reply:
Maybe a filter implemented on the Candidate nodes that can be generalized?

Reply:
So we build the build the maintenance map once for the deprovisioning controller loop, and then one more for validation on consolidation, which is necessary. I don't think that's too much, as it's also rather inexpensive.